source/DSCResources/DSC_CMSoftwareUpdatePointComponent/DSC_CMSoftwareUpdatePointComponent.psm1

```diff
@@ -0,0 +1,1517 @@
+        $EnableThirdPartyUpdates,
+
+        [Parameter()]
+        [ValidateRange(5,9999)]
+        [UInt32]
+        $FeatureUpdateMaxRuntimeMins,
+
+        [Parameter()]
+        [ValidateRange(5,9999)]
+        [UInt32]
+        $NonFeatureUpdateMaxRuntimeMins
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode
+
+    try
+    {
+        $nonChildParams = @('LanguageSummaryDetails','LanguageSummaryDetailsToInclude','LanguageSummaryDetailsToExclude','Products','ProductsToInclude','ProductsToExclude','UpdateClassifications',
+        'UpdateClassificationsToInclude','UpdateClassificationsToExclude','ContentFileOption','DefaultWsusServer','EnableCallWsusCleanupWizard','EnableSyncFailureAlert','EnableSynchronization',
+        'ImmediatelyExpireSupersedence','ImmediatelyExpireSupersedenceForFeature','SynchronizeAction','UpstreamSourceLocation','WaitMonth','WaitMonthForFeature','EnableThirdPartyUpdates',
+        'EnableManualCertManagement','FeatureUpdateMaxRuntimeMins','NonFeatureUpdateMaxRuntimeMins','ScheduleType','RecurInterval','MonthlyWeekOrder','DayOfWeek','DayOfMonth','Start')
+
+        $availableLangs = @('Arabic','Bulgarian','Chinese (Simplified, PRC)','Chinese (Traditional, Hong Kong S.A.R.)','Chinese (Traditional, Taiwan)','Croatian','Czech','Danish','Dutch','English',
+        'Estonian','Finnish','French','German','Greek','Hebrew','Hindi','Hungarian','Italian','Japanese','Korean','Latvian','Lithuanian','Norwegian','Polish','Portuguese','Portuguese (Brazil)',
+        'Romanian','Russian','Serbian','Slovak','Slovenian','Spanish','Swedish','Thai','Turkish','Ukrainian')
+
+        $langs = $LanguageUpdateFiles + $LanguageUpdateFilesToInclude + $LanguageUpdateFilesToExclude + $LanguageSummaryDetails + $LanguageSummaryDetailsToInclude + $LanguageSummaryDetailsToExclude
+        $cats = $Products + $ProductsToInclude + $ProductsToExclude + $UpdateClassifications + $UpdateClassificationsToInclude + $UpdateClassificationsToExclude
+
+        foreach ($lang in $langs.Where({ $_ -ne $null }))
+        {
+            if ($availableLangs -notcontains $lang)
+            {
+                throw ($script:localizedData.InvalidLangs -f $lang)
+            }
+        }
+
+        foreach ($cat in $cats.Where({ $_ -ne $null }))
+        {
+            if ($State.AvailableCats -notcontains $cat)
+            {
+                throw ($script:localizedData.InvalidCats -f $cat)
+            }
+        }
+
+        if ($state.ChildSite -eq $true)
+        {
+            foreach ($child in $nonChildParams)
+            {
+                if ($PSBoundParameters.ContainsKey($child))
+                {
+                    Write-Warning -Message ($script:localizedData.ChildBadParams -f $child)
+                }
+            }
+
+            if ($LanguageUpdateFiles -or $LanguageUpdateFilesToInclude -or $LanguageUpdateFilesToExclude)
```

Comment:
Do we want to give the ability to Null out languages?

Reply:
fixed.